spec/unit/cds_synchronizer_spec.rb

```diff
@@ -106,40 +107,36 @@
         expect { described_class.apply }.to raise_error(TariffSynchronizer::FailedUpdatesError)
       end
     end
+
+    context 'with xi service' do
+      xit 'raises a wrong environment error' do
+        allow(TradeTariffBackend).to receive(:service).and_return('xi')
+        expect { described_class.apply }.to raise_error TariffSynchronizer::WrongEnvironmentError
+      end
+    end
   end
 
-  describe 'check sequence of Taric daily updates' do
-    let(:applied_sequence_number) { 123 }
+  describe '.rollback' do
+    let(:rollback_attributes) { attributes_for :rollback }
 
     before do
-      create(:taric_update, :applied, example_date: Time.zone.yesterday, sequence_number: applied_sequence_number)
-      create(:taric_update, :pending, example_date: Time.zone.today, sequence_number: pending_sequence_number)
-
-      allow(TradeTariffBackend).to receive(:with_redis_lock)
-      allow(TradeTariffBackend).to receive(:uk?).and_return(false)
+      allow(TradeTariffBackend).to receive(:service).and_return('uk')
```

Comment:
Should mock `use_cds?` as well

Reply:
@jebw same here lines 79 & 80 are creating a CDSUpdate class so use_cds? would be redundant here.

app/lib/tariff_synchronizer/cds_update.rb

Reply:
Likewise - you're right its unneeded so just ignore this point :+1: